cmd/sops/common/common.go

```diff
@@ -136,3 +140,178 @@ func DefaultStoreForPath(path string) Store {
 	}
 	return &json.BinaryStore{}
 }
+
+const KMS_ENC_CTX_BUG_FIXED_VERSION = "3.3.0"
+
+func DetectKMSEncryptionContextBug(tree *sops.Tree) (bool, error) {
+	versionCheck, err := version.AIsNewerThanB(KMS_ENC_CTX_BUG_FIXED_VERSION, tree.Metadata.Version)
+	if err != nil {
+		return false, err
+	}
+
+	if versionCheck {
+		_, _, key := GetKMSKeyWithEncryptionCtx(tree)
+		if key != nil {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func GetKMSKeyWithEncryptionCtx(tree *sops.Tree) (kgndx int, keyndx int, key *kms.MasterKey) {
```

Comment:
This wouldn't handle files with more than one KMS key and encryption context, right? So the user would need to decrypt once for every key they have that would trigger the bug? If we were to support handling more than one key at a time, I'd wrap the current return values in a struct and return a slice of that struct.

Reply:
I thought so as well, but it actually does work with files with more than one. Since decrypting the datakey once works, we then re-encrypt it for all keys, including any other KMS keys with encryption context.

Reply:
Oh, right, we only need to recover the data key from one. That makes sense!